server/utils/resident/placement.test.ts

```diff
@@ -0,0 +1,158 @@
+import { TransferReason } from '@approved-premises/api'
+import { placementSideNavigation, previousApStaysCards, placementDetailsCards, applicationCards } from './placement'
+import { PlacementSubTab } from './index'
+import { cas1SpaceBookingFactory } from '../../testutils/factories'
+
+describe('placement', () => {
+  describe('placementSideNavigation', () => {
+    it('builds the side navigation for the placement tab', () => {
+      const subTab: PlacementSubTab = 'placement-details'
+      const placement = cas1SpaceBookingFactory.build()
+      const crn = 'B1663R'
+      const basePath = `/manage/resident/${crn}/placement/${placement.id}/placement/`
+
+      expect(placementSideNavigation(subTab, crn, placement.id)).toEqual([
+        {
+          active: true,
+          href: `${basePath}placement-details`,
+          text: 'Placement details',
+        },
+        {
+          active: false,
+          href: `${basePath}application`,
+          text: 'Application',
+        },
+        {
+          active: false,
+          href: `${basePath}previous-ap-stays`,
+          text: 'Previous AP stays',
+        },
+      ])
+    })
+  })
+
+  describe('previousApStaysCards', () => {
+    it('should render the previous AP stays cards with actual dates when available', () => {
+      const bookings = cas1SpaceBookingFactory.buildList(2, {
```

Comment:
The `cas1SpaceBookingFactory` has a bunch of pre-defined states that you can use. You probably don't need to hand-spin here.